src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/References/ReferenceCleanupService.cs

```diff
@@ -0,0 +1,252 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using EnvDTE;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Exceptions;
+using Microsoft.VisualStudio.ProjectSystem.VS.References.Roslyn;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.References
+{
+    [Export(typeof(IReferenceCleanupService))]
+    internal partial class ReferenceCleanupService : IProjectSystemReferenceCleanupService
+    {
+        ReferenceHandler _projectReferenceHandler = new ProjectReferenceHandler();
+        ReferenceHandler _packageReferenceHandler = new PackageReferenceHandler();
+        ReferenceHandler _assemblyReferenceHandler = new AssemblyReferenceHandler();
+        ReferenceHandler _sdkReferenceHandler = new SdkReferenceHandler();
```

Comment:
If these were a single `Dictionary<string, ReferenceHandler> _handlerByReferenceType` instead:

- `GetProjectSnapshotAsync` could be a `foreach` over `Values`
- `GetReferences` could be a `Values.SelectMany(...).ToList()`
- `FindReferenceHandler` could be a `TryGetValue`